File: 2020/csharp/Aoc2020.Tests/Day06Tests.cs
using System.Linq;
using System.Collections.Generic;
using Xunit;

namespace Aoc2020.Tests
{
    public class Day06Tests
    {
        [Fact]
        public void Stars()
        {
            var run = new Day06();
            Assert.Equal("6768", run.Result1());
            Assert.Equal("3489", run.Result2());
        }
    }
}

File: 2016/fsharp/Aoc2016.Tests/Day06.Tests.fs
module Day06.Tests

open Xunit
open Utils
open Day06

let input =
    """
    eedadn
    drvtee
    eandsr
    raavrd
    atevrs
    tsrnev
    sdttsa
    rasrtv
    nssdts
    ntnada
    svetve
    tesnvt
    vntsnd
    vrdear
    dvrsen
    enarar
    """
    |> String.trim
    |> String.split '\n'
    |> Array.map String.trim

[<Fact>]
let ``Day06 corrected works`` () =
    Assert.Equal("easter", input |> corrected Seq.maxBy)
    Assert.Equal("advent", input |> corrected Seq.minBy)

[<Fact>]
let ``Day06 Stars`` () =
    try
        Assert.Equal("qoclwvah", job1 ())
        Assert.Equal("ryrgviuv", job2 ())
    with :? System.NotImplementedException ->
        ()
